src/NHSD.BuyingCatalogue.Database/PostDeployment/InsertAssociatedServices.sql

```diff
@@ -0,0 +1,40 @@
+﻿DECLARE @emptyGuid AS uniqueidentifier = '00000000-0000-0000-0000-000000000000';
+DECLARE @now AS datetime = GETUTCDATE();
+
+DECLARE @declarative AS int = (SELECT ProvisioningTypeId FROM dbo.ProvisioningType WHERE [Name] = 'Declarative');
+DECLARE @onDemand AS int = (SELECT ProvisioningTypeId FROM dbo.ProvisioningType WHERE [Name] = 'OnDemand');
+
+DECLARE @flat AS int = (SELECT CataloguePriceTypeId FROM dbo.CataloguePriceType WHERE [Name] = 'Flat');
+DECLARE @tiered AS int = (SELECT CataloguePriceTypeId FROM dbo.CataloguePriceType WHERE [Name] = 'Tiered');
+
+DECLARE @hour AS uniqueidentifier = (SELECT PricingUnitId FROM dbo.PricingUnit WHERE [Name] = 'hour');
+DECLARE @course AS uniqueidentifier = (SELECT PricingUnitId FROM dbo.PricingUnit WHERE [Name] = 'course');
+
+DECLARE @associatedServiceItemType AS int = (SELECT CatalogueItemTypeId FROM dbo.CatalogueItemType WHERE [Name] = 'Associated Service');
+DECLARE @publishedStatus AS int = (SELECT Id FROM dbo.PublicationStatus WHERE [Name] = 'Published');
+
+DECLARE @gbp AS char(3) = 'GBP';
+
+DECLARE @associatedServiceId AS varchar(14);
+
+IF UPPER('$(INSERT_TEST_DATA)') = 'TRUE' AND NOT EXISTS (SELECT * FROM dbo.AssociatedService)
+BEGIN
+    SET @associatedServiceId = '100000-S-001';
+
+    INSERT INTO dbo.CatalogueItem(CatalogueItemId, CatalogueItemTypeId, [Name], SupplierId, PublishedStatusId, Created)
+         VALUES (@associatedServiceId, @associatedServiceItemType, 'Really Kool associated service', '100000', @publishedStatus, @now);
+
+    INSERT INTO dbo.AssociatedService (AssociatedServiceId, [Description], OrderGuidance, LastUpdated, LastUpdatedBy) 
+         VALUES (@associatedServiceId, 'Really Kool associated service', NULL, @now, @emptyGuid);
+
+    INSERT INTO dbo.CataloguePrice (CatalogueItemId, ProvisioningTypeId, CataloguePriceTypeId, PricingUnitId, TimeUnitId, CurrencyCode, LastUpdated, Price)
+         VALUES (@associatedServiceId, @declarative, @flat, @course, NULL, @gbp, @now, 99.99),
+                (@associatedServiceId, @onDemand, @tiered, @hour, NULL, @gbp, @now, NULL);
```

Comment:
I thought we weren't doing tiered any more?

Reply:
Sorry, but I have no idea. Does that mean the only valid options for associated services are flat/declarative and flat/on-demand?

Reply:
I think they're still valid but are descoped. so whilst adding we can add the data in we aren't building the functionality to use the tiered prices so we won't be able to use that data. @JoshuaRobertsBJSS is that right?
Plus i need flat/declarative and flat/on-demand rather than tiered to test the ACs

Reply:
Tiered pricing, in general, has been moved out of scope. However on the price selection screen, we kept tiered pricing in the ACs for additional services, as we had already done it for the catalogue solutions. So you can view a tiered price, and if you select it - it blows up, that's fine. But you can still see it on the 'select price' screen. 

The idea was that we keep the additional service story the same as the catalogue solution story, and so the tiered pricing ACs remain, even though the ability to save a tiered price is not in scope. 

We're ok with this - because no irl solution currently has a tiered price, when one is introduced, we know we need to bring tiered prices back into scope. 

Does that make sense?

Reply:
I've added a flat/on-demand price and moved the tiered pricing to a different associated service, so Really Kool will have two associated services.